app/src/main/java/sleepyweasel/purplefluffernutter/MovieContentProvider.java

```diff
@@ -0,0 +1,64 @@
+package sleepyweasel.purplefluffernutter;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MovieContentProvider {
+
+    private int nextId;
+
+    private HashMap<Integer,MovieEntry> movies;
+
+    private MovieContentProvider() {
+        movies = new HashMap<>();
+        nextId = 0;
+    }
+
+    public static MovieContentProvider getInstance() {
+        return MovieContentProviderHolder.instance;
+    }
+
+    public boolean isEmpty() {
+        return movies.isEmpty();
+    }
+
+    public void addMovie(MovieEntry entry) {
+        movies.put(nextId, entry);
+        nextId++;
+    }
+
+    public int size() {
+        return movies.size();
+    }
+
+    public MovieEntry getEntry(int id) {
+        return movies.get(id);
+    }
+
+    public MovieEntry getEntryByTitle(String movieTitle) {
+        for (Map.Entry<Integer,MovieEntry> entry : movies.entrySet()) {
+            MovieEntry movie = entry.getValue();
+            if (movie.getTitle().equals(movieTitle))
+                return movie;
+        }
+        return null;
+    }
+
+    public void clear() {
+        movies.clear();
+        nextId = 0;
+    }
+
+    public ArrayList<String> getArrayList() {
```

Comment:
hm.. rzeczywiście jest beznadziejna. Generalnie chodzi o to, że w tym ListAdapterze, który tworzę w MovieListFragment trzeba mu podać jako argument listę, żeby sobie przy jej pomocy spopulował widok. Jakiś pomysł na lepszą nazwę?